libs/hdf_converters/src/aws-config-mapper.ts

```diff
@@ -0,0 +1,345 @@
+    if (result.ComplianceType === 'COMPLIANT') {
+      return ExecJSON.ControlResultStatus.Passed;
+    } else if (result.ComplianceType === 'NON_COMPLIANT') {
+      return ExecJSON.ControlResultStatus.Failed;
+    } else {
+      return ExecJSON.ControlResultStatus.Skipped;
+    }
+  }
+  private getMessage(
+    result: EvaluationResult,
+    status: ExecJSON.ControlResultStatus
+  ): string {
+    if (status === ExecJSON.ControlResultStatus.Failed) {
+      return `${result.EvaluationResultIdentifier}: ${
+        result.EvaluationResultIdentifier?.EvaluationResultQualifier
+      }: ${result.Annotation || 'Rule does not pass rule compliance'}`;
+    } else {
+      return '';
+    }
+  }
+  private addComplianceToConfigRules(configRules: ConfigRule[]): ConfigRule[] {
+    const mappedComplianceInfo = this.fetchAllComplianceInfo(configRules);
+    return configRules.map((rule) => {
+      return _.set(
+        rule,
+        'compliance',
+        _.get(
+          mappedComplianceInfo,
+          `${rule.ConfigRuleName}.compliance.complianceType`
+        )
+      );
+    });
+  }
+  private fetchAllComplianceInfo(configRules: ConfigRule[]): unknown[] {
+    let complianceResults: ComplianceByConfigRule[] = [];
+    _.chunk(configRules, 25).forEach(async (group) => {
+      const configRuleNames = group.map(
+        (element) => element.ConfigRuleName || ''
+      );
+      const params: DescribeComplianceByConfigRuleCommandInput = {
+        ComplianceTypes: [
+          'COMPLIANT',
+          'NON_COMPLIANT',
+          'NOT_APPLICABLE',
+          'INSUFFICIENT_DATA'
+        ],
+        ConfigRuleNames: configRuleNames,
+        NextToken: ''
+      };
+      const response = await this.configService.describeComplianceByConfigRule(
+        params
+      );
+      if (response.ComplianceByConfigRules !== undefined) {
+        complianceResults = complianceResults.concat(
+          response.ComplianceByConfigRules
+        );
+      }
+    });
+    return complianceResults.map((element) =>
+      Object.fromEntries([[element.ConfigRuleName, element]])
+    );
+  }
+  // eslint-disable-next-line @typescript-eslint/ban-types
+  private hdfTags(configRule: ConfigRule): Record<string, unknown> {
+    let result = {};
+    const sourceIdentifier = configRule.Source?.SourceIdentifier;
+    result = _.set(result, 'nist', []);
+    let defaultMatch: string[] | null = [];
+    if (sourceIdentifier !== undefined) {
+      defaultMatch = AWS_CONFIG_MAPPING.nistFilter([sourceIdentifier]);
+    }
+    if (Array.isArray(defaultMatch) && defaultMatch.length !== 0) {
+      result = _.set(
+        result,
+        'nist',
+        _.get(result, 'nist').concat(defaultMatch)
+      );
+    }
+
+    //Deal with custom mapping later
+    // custom_mapping_match = @custom_mapping[source_identifier]
+
+    // result['nist'] += custom_mapping_match['NIST-ID'].split('|').map { | name | "#{name} (user provided)"} unless custom_mapping_match.nil ?
```

Comment:
Is this comment still relevant? If the code is not necessary then it should be removed.

Reply:
Yes, I need to ask how the user would pass in a custom mapping in the first place. In addition, I'm unable to test the mapper to begin with, so I'm just sort of waiting on an access key at this point in time.